src/status-codes.js

```diff
@@ -0,0 +1,51 @@
+// @flow
+import type {NotFoundStatus, OKStatus, RedirectStatus} from "./types.js";
+
+const NotFoundImpl: NotFoundStatus = {
+    get code(): 404 {
+        return 404;
+    },
+};
+
+/**
+ * Create a 404 NotFound status.
+ */
+export const notFound = (): NotFoundStatus => NotFoundImpl;
```

Comment:
In what scenario would the server know to generate a 404 error? I feel like that needs to be left up to the rendering logic to handle. It seems much more likely to me that we need to have a 500 status that we can return if there's an exception during rendering - there render-gateway isn't going to know if some URL does, or doesn't, exist, right?

Reply:
At some point, the webapp code has to signal back to the gateway what code to return, including 404s where our routing didn't match the requested route. This will be part of the API, but yeah, we can certainly simplify how that will work, though I worry about allowing just anything - I wanted to build in some rigor so that it's easy to get it right.

That said, perhaps we can do that by providing a different set of helpers to build a render result that encourage supplying the correct headers.

Reply:
IMO, I'm fine deferring the question of rigor until much later - I think for now most of this logic is going to be managed in PageSettings anyway so we can just pass back the status code + headers + body and keep render-gateway nice and simple!